canonical-web-and-design/vanilla-framework
# Integrate examples into docs

As discussed with @deadlight, I've done the following:

- Converted the docs in `/docs` into a Jekyll site format
- Remove superfluous stuff from `/docs` for the old documentation-builder stuff
- Removed `/en/`, and added a redirect
- Moved `/examples` into the `/docs` folder
- Switched the docs pages to point to these local examples
- Updated the root-level `package.json` so that `./run` will serve the documentation site (which now includes examples)
- Added scripts to `package.json` for `build-docs-site`, `build-docs-css` and `build-docs`
- Updated the README.md to explain how documentation now works
- Updated the `/docs/Dockerfile` to build the production image for docs.vanillaframework.io based off this new structure
- Updated `/examples/` index page to use the same styling as the rest of the docs.
- The right-hand "Contents" menu is now generated client-side with JavaScript

Fixes https://github.com/canonical-web-and-design/base-squad/issues/511

QA
--

`./run`

- Go to http://0.0.0.0:8101/, check it works.
- Browse around a bit
- Check inline examples work
- Go to http://0.0.0.0:8101/examples/, check the examples work
- Check you see the "toggle" button when viewing examples directly, but not when viewing examples on a documentation page

Now test the Docker image:

``` bash
cd docs/
docker build --build-arg COMMIT_ID=fake --tag docs .
docker run -ti -p 8901:80 docs
```

- Go to http://0.0.0.0:8901, check it works.
- Check old URLs (e.g. `http:0.0.0.0:8901/en/patterns/grid`) redirect to new ones (e.g. `http:0.0.0.0:8901/patterns/grid`)
